config/cloud.cfg.tmpl

```diff
@@ -85,6 +85,9 @@ cloud_init_modules:
  - seed_random
 {% endif %}
  - bootcmd
+{% if variant in ["ubuntu"] %}
+ - ubuntu_autoinstall
```

Comment:
I think this needs to be further down in the boot process. When I tried this, the boot hung for a minute or two, and I eventually got this in the logs:
```
2022-07-15 17:47:47,471 - util.py[DEBUG]: Running module ubuntu_autoinstall (<module 'cloudinit.config.cc_ubuntu_autoinstall' from '/usr/lib/python3/dist-packages/cloudinit/config/cc_ubuntu_autoinstall.py'>) failed
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/cloudinit/config/modules.py", line 231, in _run_modules
    ran, _r = cc.run(
  File "/usr/lib/python3/dist-packages/cloudinit/cloud.py", line 67, in run
    return self._runners.run(name, functor, args, freq, clear_on_fail)
  File "/usr/lib/python3/dist-packages/cloudinit/helpers.py", line 185, in run
    results = functor(*args)
  File "/usr/lib/python3/dist-packages/cloudinit/config/cc_ubuntu_autoinstall.py", line 82, in handle
    snap_list, _ = subp(["snap", "list"])
  File "/usr/lib/python3/dist-packages/cloudinit/subp.py", line 335, in subp
    raise ProcessExecutionError(
cloudinit.subp.ProcessExecutionError: Unexpected error while running command.
Command: ['snap', 'list']
Exit code: 1
Reason: -
Stdout: 
Stderr: error: cannot list snaps: cannot communicate with server: Get http://localhost/v2/snaps: dial unix /run/snapd.socket: connect: no such file or directory
```

Maybe go under the snap module instead?

Reply:
Ohh right. I had forgotten about that dependency... you are right on a couple counts:
1. and yes there'll be snap seeding races we don't want to cope with too on some cloud images
2. userdata may want to install the subiquity snap to allow autoinstall in some corner cases